spharpy/classes/ambisonics.py

```diff
@@ -0,0 +1,220 @@
+            (maxN is only supported up to 3rd order)
+        channel_convention : str
+            Channel ordering convention, either ``'acn'`` or ``'fuma'``.
+            The default is ``'acn'``.
+            (FuMa is only supported up to 3rd order)
+        condon_shortley : bool or str, optional
+            Whether to include the Condon-Shortley phase term. If ``True`` or
+            ``'auto'``, Condon-Shortley is included, if ``False`` it is not
+            included. The default is ``'auto'``.
+        n_samples : int, optional
+            Number of samples of the time signal. Required if domain is
+            ``'freq'``. The default is ``None``, which assumes an even number
+            of samples if the data is provided in the frequency domain.
+        domain : ``'time'``, ``'freq'``, optional
+            Domain of data. The default is ``'time'``
+        fft_norm : str, optional
+            The normalization of the Discrete Fourier Transform (DFT). Can be
+            ``'none'``, ``'unitary'``, ``'amplitude'``, ``'rms'``, ``'power'``,
+            or ``'psd'``. See :py:func:`~pyfar.dsp.fft.normalization` and [#]_
+            for more information. The default is ``'none'``, which is typically
+            used for energy signals, such as impulse responses.
+        comment : str
+            A comment related to `data`. The default is ``None``.
+
+        References
+        ----------
+        .. [#] E.G. Williams, "Fourier Acoustics", (1999), Academic Press
+        .. [#] B. Rafely, "Fundamentals of Spherical Array Processing", (2015),
+               Springer-Verlag
+        .. [#] F. Zotter, M. Frank, "Ambisonics A Practical 3D Audio Theory
+               for Recording, Studio Production, Sound Reinforcement, and
+               Virtual Reality", (2019), Springer-Verlag
+
+        """
+    def __init__(
+            self,
+            data,
+            sampling_rate,
+            n_max,
+            basis_type,
+            normalization,
+            channel_convention,
+            condon_shortley='auto',
+            n_samples=None,
+            domain='time',
+            fft_norm='none',
+            comment="",
+            is_complex=False):
+        """
+        Create SphericalHarmonicSignal with data, and sampling rate.
+        """
+        # set n_max
+        if (n_max < 0) or (n_max % 1 != 0):
+            raise ValueError("n_max must be a positive integer")
+        if not data.shape[-2] >= (n_max + 1) ** 2:
+            raise ValueError('Data has to few sh coefficients for '
+                             f'{n_max=}. Highest possible n_max is '
+                             f'{int(np.sqrt(data.shape[-2]))-1}')
+        self._n_max = int(n_max)
+
+        # set basis_type
+        if basis_type == 'complex' and not is_complex:
+            raise ValueError('Data are real-valued while '
+                             'spherical harmonics bases are complex-valued.')
```

Comment:
Not sure if we should handle it this way. If the time data before the SH transform is real-valued, the time data after the SH transform will be real values as well. If the time data before the SH-transform ist complex valued, the time data after the SH transfrom will be complex-valued even if using a real-valued SH basis. I would suggest to not check this and use the definition of `is_complex` from the Sigbal class. Most likely there will never be the case of a complex SH Signal in this sense, but I don't see any harm in keeping the possibility, if we clearly distinguish between complex valued signals and real- and complex valued SH-bases.